PowerValidatedSolutions.psm1

```diff
@@ -48145,6 +48145,383 @@ Function Test-WMSubnetInput {
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $username,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $password,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $csrFilePath,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $certDir,            
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $certificateTemplate,
+        [Parameter (Mandatory = $false)] [Switch] $getCArootCert
+    )
+
+    # Validate inputs
+    if (-Not (Test-Path -Path $certDir)) {
+        Write-Error "Error: $certDir directory does not exist." -ErrorAction Stop
+    }
+    if (-Not (Test-Path -Path $csrFilePath)) {
+        Write-Error "Error: $csrFilePath file not found." -ErrorAction Stop
+    }
+    $fileName = Split-Path -Path $csrFilePath -LeafBase
+    $crtFileName = $fileName + ".crt"
+    $crtFilePath = Join-Path -Path $certDir -ChildPath $crtFileName
+    if (Test-Path -Path $crtFilePath) {
+        Write-Error "Error: $crtFilePath file exists.  Please remove or rename the existing file." -ErrorAction Stop
+    }
+
+    $securePass = ConvertTo-SecureString -String $password -AsPlainText -Force
+    $domainCreds = New-Object System.Management.Automation.PSCredential ($username, $securePass)
+    
+    Try {
+        $respond = Invoke-WebRequest -Uri "https://$caFqdn/certsrv/certrqxt.asp" -Credential $domainCreds -Method Get -SkipCertificateCheck
+    } catch {
+        Debug-ExceptionWriter -object $_
+    }
+
+    if (-Not ($respond.Content -match ".*Microsoft RSA SChannel Cryptographic Provider.*$certificateTemplate.*")) {
+        Write-Error "The Microsoft Certificate Template $certificateTemplate not found." -ErrorAction Stop
+    } else {
+        $CertAttributes = "CertificateTemplate:$certificateTemplate"
+    }
+
+    if ($getCArootCert.IsPresent) {
+        $createRandomCounter = 0
+        for ($createRandomCounter = 0; $createRandomCounter -lt 8; $createRandomCounter++) {
+            $randomString = -join (((48..57)+(65..90)+(97..122)) * 80 |Get-Random -Count 6 |%{[char]$_})
+            $rootCaFilePathTemp = Join-Path -path $certDir -ChildPath "$fileName-$randomString.p7b"
+            if (-Not(Test-Path -Path $rootCaFilePathTemp)) {
+				Write-Output "1" | Set-Content $rootCaFilePathTemp
+				if(Test-Path -Path $rootCaFilePathTemp) {
+					Remove-Item $rootCaFilePathTemp
+					break;
+				}
+			} else {
+				if($createRandomCounter -gt 6) {
+					Write-Error "Error: Unable to create temporary files in directory $certDir." -ErrorAction Stop
+					Exit
+				}
+			}
+		}
+        $rootCaFilename = "$caFqdn-rootCA.pem"
+        $rootCaFilePath = Join-Path -Path $certDir -ChildPath $rootCaFilename
+        if (Test-Path -Path $rootCaFilePath) {
+            Write-Error "Error: $rootCaFilePath file exists.  Please remove or rename the existing file." -ErrorAction Stop
```

Comment:
```suggestion
            Write-Error "$rootCaFilePath file exists.  Please remove or rename the existing file." -ErrorAction Stop
```

Reply:
Addressed in commit 53039acbaaffafaf219274cac1a3684735130f53